arkfilepicker/src/main/java/space/taran/arkfilepicker/folders/RootsScan.kt

```diff
@@ -0,0 +1,41 @@
+package space.taran.arkfilepicker.folders
+
+import android.util.Log
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ensureActive
+import kotlinx.coroutines.withContext
+import space.taran.arkfilepicker.FileUtils
+import space.taran.arkfilepicker.LogTags
+import space.taran.arkfilepicker.arkFolder
+import java.nio.file.Path
+import java.util.LinkedList
+import java.util.Queue
+import kotlin.io.path.exists
+import kotlin.io.path.isDirectory
+import kotlin.io.path.listDirectoryEntries
+
+class ArkRootsScan {
```

Comment:
Scanning in Filepicker should work like this:
1. Filepicker should start in classic mode.
2. If user switches to ARK mode, we check presence of the roots file.
3. If the roots file exists, we just take roots from it.
4. Otherwise, we suggest scanning to the user.
5. We do scanning, write to the roots file.

Also, Navigator should call the code from Filepicker in order to avoid duplication.

@ShubertMunthali @mdrlzy does it work like this now?

Reply:
If the user doesn't switch to ARK mode, scanning should not happen and should not be suggested.

Reply:
Filepicker should also remember in which mode it was used last time.
If last time was ARK mode, the user should start in ARK mode next time.
If last time was classic mode, the user starts in classic mode.

Reply:
Yes it works that way now, only that user cannot choose to use ARK-Mode or not

Reply:
What about this? We need new preference for this, I guess.

> Filepicker should also remember in which mode it was used last time.
> If last time was ARK mode, the user should start in ARK mode next time.
> If last time was classic mode, the user starts in classic mode.

Reply:
Yes